src/features/Gw2/actions.js

```diff
@@ -66,7 +66,7 @@ export function generateActions (resourceName, getResource, afterGet) {
       // We filter them, then pass them along to the service (for example lib/gw2/readCalculatedItemStats)
       // If calculatedId exists, use that.
       const actualId = +(typeof id === 'object' ? (id.calculatedId || id.id) : id);
-      const isValidId = actualId && actualId !== -1;
+      const isValidId = (id === 'all') || (actualId && actualId !== -1);
```

Comment:
is this really needed though? the current logic `actualId && actualId !== -1` would return true for the string `'all'`.

```
> 'all' && 'all' !== -1
true
```

Reply:
actualId was not 'all', it was NaN. Maybe because of the + right after the assignment equals?

Reply:
ah, good catch. that'll do it.

can you add a test case for this pls? the reducer/action logic is pretty well tested (compared to the rest of the code base.. haha)